Adding new information to the HoloViews About Page (#4370)
holoviz/holoviews: doc/about.rst

HoloViews is an `open-source <https://github.com/holoviz/holoviews/blob/master/LICENSE.txt>`_ Python library designed to make data analysis and visualization seamless and simple.  With HoloViews, you can usually express what you want to do in very few lines of code, letting you focus on what you are trying to explore and convey, not on the process of plotting.

HoloViews is part of the `HoloViz <https://holoviz.org>`_ family of tools. The `holoviz.org <https://holoviz.org>`_ website shows how to use HoloViews together with other libraries to solve complex problems, with detailed tutorials and examples. You can see a variety of projects using HoloViews at `examples.pyviz.org <https://examples.pyviz.org>`_, and you can compare HoloViews to other available tools at `pyviz.org <https://pyviz.org>`_.

> If you have any questions or usage issues visit the `HoloViews Discourse <https://discourse.holoviz.org/c/holoviews/>`_ site. If you are interested in contributing to HoloViews development to help address some of the `open issues <https://github.com/holoviz/holoviews/issues>`_, see our `developer instructions https://holoviews.org/index.html` to set up your development environment.

Reviewer: ```suggestion
If you have any questions or usage issues visit the `HoloViews Discourse <https://discourse.holoviz.org/c/holoviews/>`_ site. If you are interested in contributing to HoloViews development to help address some of the `open issues <https://github.com/holoviz/holoviews/issues>`_, see our `developer instructions <https://holoviews.org/index.html>`_ to set up your development environment.
```

> If you have any questions or usage issues visit the `HoloViews Discourse <https://discourse.holoviz.org/c/holoviews/>`_ site. If you are interested in contributing to HoloViews development to help address some of the `open issues <https://github.com/holoviz/holoviews/issues>`_, see our `developer instructions <https://holoviews.org/index.html>`_ to set up your development environment.

Reviewer: Actually, that's not a link to the developer instructions; what should it really be?

Reviewer 2: Oh missed that, currently the only project that has developer instructions is Panel.

Reviewer: I think we should simplify Panel's instructions and put them at holoviz.org for all the rest of the projects to refer to.

Reviewer 2: Sure I'm sure we can pull stuff out but Panel will definitely need to keep it's own developer docs and so does HoloViews tbh.

Author: Ok, should I take out the developer instructions for the time being on this page and the rest of them?

Reviewer 2: I see relatively little value in shared developer docs tbh, all the bits that can be shared will rarely change and therefore can easily be duplicated without becoming outdated while the rest should explain specifics about the project, such the the architecture, design decisions and organization of the project.

Reviewer: The problems with having separate developer docs are (a) people don't develop on just one of our libraries; they very often start with one and then make changes to another, and shouldn't have to throw away their environments when they do so, and (b) our various instructions and our own developer approaches are mutually contradictory and confusing.  I think we should work out one main shared set of instructions and agree on them before propagating them anywhere. Personally, I'd be fine with keeping this PR open until the shared link target is available.

Reviewer: I've had _numerous_ people express confusion about just what they are meant to do when making a PR on more than one project, yet we each (I think) maintain shared development environments for all HoloViz projects we work on, and I think we should tell people how to do that!

Reviewer 2: >people don't develop on just one of our libraries; they very often start with one and then make changes to another, and shouldn't have to throw away their environments when they do so,

I know very, very few people who have made changes to more than one project.

>our various instructions and our own developer approaches are mutually contradictory and confusing

We don't have developer docs but I can agree that some of the install instructions may be contradictory.

>I think we should work out one main shared set of instructions and agree on them before propagating them anywhere.

Oh for sure, I'm just not sure all projects will be pointing to that shared target as it'd be weird to have a developer section in the docs of a project which then just ends up linking elsewhere for the most basic instructions.

Reviewer: Here I'm not even talking about developer docs as the source of confusion, but what we tell all the people who have asked us on gitter what to do; each one of them has gotten different instructions, and very many of them have come back and said they are confused.  So even if a particular project doesn't point back to the holoviz main developer docs, those docs will serve as a consistent reference to keep our instructions (whether on websites or in chats) from confusing everyone.

Reviewer: Basically, I just want us to agree what to tell people, then tell people that. Beyond that is just details. :-)

Reviewer 3: It'd be indeed very useful to have more guidance about how to contribute to each project's code base! On the one hand I tend to believe it should not sit on holoviz but rather on each project's website because basically I rarely visit the holoviz website (it's not where the relevant content for me lives, like API doc and usage). On the other hand, some of it could be holoviz if you agree on a pattern that is general enough.

If you do want more people to contribute to your projects, I think a "Contributing to project_name" page is a must (you'll find that on a lot of open source projects). It'd not only describe how to contribute to the code (dev setup, git/github workflow, code style, etc.), but also how to contribute to the community (discourse, twitter, report issues, etc.) and most importantly how to contribute to the docs itself (updating notebooks, improving docstrings, etc.).

You already have a great documentation, you would make it even better ;) (I'm aware this is a big undertaking...)

> If you have any questions or usage issues visit the `HoloViews Discourse <https://discourse.holoviz.org/c/holoviews/>`_ site. If you are interested in contributing to HoloViews development to help address some of the `open issues <https://github.com/holoviz/holoviews/issues>`_, see our `developer instructions <https://holoviews.org/index.html>`_ to set up your development environment.

Reviewer: ```suggestion
If you have any questions or usage issues visit the `HoloViews Discourse <https://discourse.holoviz.org/c/holoviews/>`_ site. If you are interested in contributing to HoloViews development to help address some of the `open issues <https://github.com/holoviz/holoviews/issues>`_, you can follow the `developer instructions for the Panel library <https://panel.holoviz.org/developer_guide/>`_ to set up your development environment.
```

> If you like HoloViews and have built something you want to share, tweet a link or screenshot of your latest creation at @HoloViews, along with any other library you used (@Panel_org, @Datashader, @BokehPlots, @Matplotlib, etc.). Thanks!

Reviewer: ```suggestion
If you like HoloViews and have built something you want to share, tweet a link or screenshot of your latest creation at @HoloViews, along with any other library you used (@HoloViz_org, @Panel_org, @Datashader, @BokehPlots, @Matplotlib, etc.). Thanks!
```
